cmd/cloud-server/service/load-balancer/topo.go

```diff
@@ -0,0 +1,895 @@
+		return nil, err
+	}
+	if len(lbMap) == 0 {
+		return &cslb.TargetTopoInfo{Match: false}, nil
+	}
+
+	// 根据条件查询监听器信息
+	lbIDs := maps.Keys(lbMap)
+	lblCond := make([]filter.RuleFactory, 0)
+	lblCond = append(lblCond, tools.RuleIn("lb_id", lbIDs))
+	lblCond = append(lblCond, req.GetLblCond()...)
+	lblMap, err := svc.getLblByCond(kt, vendor, lblCond)
+	if err != nil {
+		logs.Errorf("get lbl by cond failed, err: %v, lblCond: %v, rid: %s", err, lblCond, kt.Rid)
+		return nil, err
+	}
+	if len(lblMap) == 0 {
+		return &cslb.TargetTopoInfo{Match: false}, nil
+	}
+
+	// 根据条件查询规则
+	lblIDs := maps.Keys(lblMap)
+	ruleCond := make([]filter.RuleFactory, 0)
+	ruleCond = append(ruleCond, tools.RuleIn("lbl_id", lblIDs))
+	ruleCond = append(ruleCond, req.GetRuleCond()...)
+	ruleMap, err := svc.getRuleByCond(kt, vendor, ruleCond)
+	if err != nil {
+		logs.Errorf("get rule by cond failed, err: %v, ruleCond: %v, rid: %s", err, ruleCond, kt.Rid)
+		return nil, err
+	}
+	if len(ruleMap) == 0 {
+		return &cslb.TargetTopoInfo{Match: false}, nil
+	}
+
+	// 根据条件查询clb和目标组关系
+	ruleIDs := maps.Keys(ruleMap)
+	tgLbRelCond := []filter.RuleFactory{tools.RuleIn("listener_rule_id", ruleIDs), tools.RuleEqual("vendor", vendor),
+		tools.RuleEqual("binding_status", enumor.SuccessBindingStatus)}
+	tgLbRels, err := svc.getTgLbRelByCond(kt, tgLbRelCond)
+	if err != nil {
+		logs.Errorf("get tg lb rel failed, err: %v, tgLbRelCond: %v, rid: %s", err, tgLbRelCond, kt.Rid)
+		return nil, err
+	}
+	if len(tgLbRels) == 0 {
+		return &cslb.TargetTopoInfo{Match: false}, nil
+	}
+	tgIDMap := make(map[string]struct{})
+	for _, tgLbRel := range tgLbRels {
+		tgIDMap[tgLbRel.TargetGroupID] = struct{}{}
+	}
+
+	// 根据条件查询目标组
+	tgIDs := maps.Keys(tgIDMap)
+	tgCond := []filter.RuleFactory{tools.RuleIn("id", tgIDs)}
+	tgMap, err := svc.getTgByCond(kt, tgCond)
+	if err != nil {
+		logs.Errorf("get tg by cond failed, err: %v, tgCond: %v, rid: %s", err, tgCond, kt.Rid)
+		return nil, err
+	}
+	if len(tgMap) == 0 {
+		return &cslb.TargetTopoInfo{Match: false}, nil
+	}
+
+	return &cslb.TargetTopoInfo{
+		Match: true, LbMap: lbMap, LblMap: lblMap, RuleMap: ruleMap, TgLbRels: tgLbRels, TgMap: tgMap,
+	}, nil
+}
+
+func buildInstWithTargetsInfo(kt *kit.Kit, clbTopoInfo *cslb.TargetTopoInfo, targets []corelb.BaseTarget,
+	instInfos []daotypeslb.ListInstInfo) ([]cslb.InstWithTargets, error) {
+
+	if clbTopoInfo == nil || len(targets) == 0 || len(instInfos) == 0 {
+		return make([]cslb.InstWithTargets, 0), nil
+	}
+
+	tgIDRelMap := make(map[string]corelb.BaseTargetListenerRuleRel)
+	for _, rel := range clbTopoInfo.TgLbRels {
+		tgIDRelMap[rel.TargetGroupID] = rel
+	}
+	ipTargetsMap := make(map[string][]cslb.TargetWithTopo)
+	for _, target := range targets {
+		tgID := target.TargetGroupID
+		targetGroup, ok := clbTopoInfo.TgMap[tgID]
+		if !ok {
+			logs.Errorf("tg id not found, tgID: %s, rid: %s", tgID, kt.Rid)
+			return nil, fmt.Errorf("tg not found, id: %s", tgID)
```

Comment:
这里返回错误就写全称吧，毕竟可能要给用户看的。下同
```suggestion
			return nil, fmt.Errorf("target group not found, id: %s", tgID)
```

Reply:
done